hedera-mirror-rest-java/src/test/java/com/hedera/mirror/restjava/controller/AllowancesControllerIntegrationTest.java

```diff
@@ -239,36 +313,94 @@ void successWithEmptyNextLink() {
+        )
+                .isInstanceOf(HttpClientErrorException.MethodNotAllowed.class)
+                .hasMessageContaining("Method Not Allowed");
+    }
+
+    @ParameterizedTest
+    @ValueSource(
+            strings = {
+                    "0.0.1000",
+                    "0.1000",
+                    "1000",
+                    "0.0x000000000000000000000000000000000186Fb1b",
+                    "0.0.0x000000000000000000000000000000000186Fb1b",
+                    "0x000000000000000000000000000000000186Fb1b",
+                    "0.0.AABBCC22",
+                    "0.AABBCC22",
+                    "AABBCC22"
+            })
+    void nftAllowancesNotFoundException(String accountId) {
+        validateErrorMessage(callUri, accountId, "No account found for the given ID",HttpClientErrorException.NotFound.class, "Not Found");
+    }
+
+    @Test
+    void nftAllowancesTokenIdValidation() {
+
+        String uriParams = "?owner=true&token.id=gt:0.0.1000&limit=1&order=asc";
+        var entity = domainBuilder.entity().persist();
+
+        // Creating nft allowances
+        var allowance1 = domainBuilder
+                .nftAllowance()
+                .customize(nfta -> nfta.owner(entity.getId()))
+                .persist();
 
         // Performing the GET operation
-        assertThrows(
-                HttpClientErrorException.NotFound.class,
-                () -> restClient.get().uri(uriParams, 1000).retrieve().body(NftAllowancesResponse.class));
+        validateErrorMessage(uriParams, EntityId.of(allowance1.getOwner()).toString(), "token.id parameter must have account.id present", HttpClientErrorException.BadRequest.class, "Bad Request");
     }
 
     @Test
-    void failWithUnsupportedMediaException() {
-        RestClient restClient = RestClient.create();
+    void nftAllowancesEntityIdRangeOperatorValidation() {
+
+        String uriParams = "?owner=true&account.id=gte:0.0.1000&token.id=ne:0.0.1000&limit=1&order=asc";
+        var entity = domainBuilder.entity().persist();
+
+        // Creating nft allowances
+        var allowance1 = domainBuilder
+                .nftAllowance()
+                .customize(nfta -> nfta.owner(entity.getId()))
+                .persist();
 
         // Performing the GET operation
-        assertThrows(
-                HttpClientErrorException.MethodNotAllowed.class,
-                () -> restClient.post().uri(callUri, "0.0.1000").retrieve().body(NftAllowancesResponse.class));
+        validateErrorMessage(uriParams, EntityId.of(allowance1.getOwner()).toString(), "Invalid range operator ne. This operator is not supported",HttpClientErrorException.BadRequest.class, "Bad Request");
     }
+
+    private void validateErrorMessage(String uriParams, String accountId, String expected, Class<? extends HttpClientErrorException> clazz, String description) {
+        assertThatThrownBy(() -> restClient.get().uri(uriParams, accountId).retrieve().body(NftAllowancesResponse.class)
```

Comment:
Prefer to decouple test invocation from error validation so it's more reusable.

```suggestion
    private void validateError(ThrowingCallable callable, Class<? extends HttpClientErrorException> clazz, String message, String description) {
        assertThatThrownBy(callable)
```